grug/types/src/response.rs

```diff
@@ -67,25 +67,23 @@ impl Response {
 
     pub fn add_event<E>(mut self, event: E) -> StdResult<Self>
     where
-        E: EventName,
+        E: TryInto<ContractEvent, Error = StdError>,
     {
-        self.subevents.push(ContractEvent::new(E::NAME, event)?);
+        self.subevents.push(event.try_into()?);
         Ok(self)
     }
 
-    pub fn add_subevent(mut self, event: ContractEvent) -> Self {
-        self.subevents.push(event);
-        self
-    }
-
-    pub fn may_add_subevent(mut self, maybe_event: Option<ContractEvent>) -> Self {
+    pub fn may_add_event<E>(mut self, maybe_event: Option<E>) -> StdResult<Self>
+    where
+        E: TryInto<ContractEvent, Error = StdError>,
```

Comment:
@Rhaki if i already have a `ContractEvent`, then i can't use it here as `maybe_event`, because `ContractEvent` itself doesn't implement `TryInto<ContractEvent, Err = StdError>`.

instead, `ContractEvent` itself implements `TryInto<ContractEvent, Err = Infallible>`, and `Inflallible` implements `Into<StdError>`. therefore you can make it work by changing the function signature to:

```rust
fn may_add_event<E>(mut self, maybe_event: Option<E>) -> StdResult<Self>
where
    E: TryInto<ContractError>,
    StdError: From<E::Error>,
{ /* ... */ }
```